utils/templates.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * The regex used to match the name of the template
+ * @type {RegExp}
+ */
+const TEMPLATE_ELEMENT_NAMES_REGEX = /<t name=['"]?([\w-]*?)['"]? ?\/? ?>/g;
+
+/**
+ * The regex used to match the translated content for the templates - open and
+ * close template element
+ * @type {RegExp}
+ */
+const TEMPLATE_OPEN_CLOSE_ELEMENT_CONTENT_REGEX = /<t name=['"]?[\w-]+['"]?\s?>(.+?)<\/t>/g;
+
+/**
+ * The regex used to match the translated content for the templates - self
+ * closing template element
+ * @type {RegExp}
+ */
+const TEMPLATE_SELF_CLOSING_ELEMENT_CONTENT_REGEX = /<t name=['"]?[\w-]+['"]? ?\/? ?>/g;
+
+/**
+ * The regex used to match br templates that are not self closed
+ * @type {RegExp}
+ */
+const TEMPLATE_BR_OPEN_ONLY_REGEX = /<t name=['"]?br+['"]? ?>/g;
+
+/**
+ * The string used to replace br templates that are not self closed.
+ */
+const TEMPLATE_BR_SELF_CLOSING_STRING = '<t name=\'br\' />';
+
+/**
+ * The error message logged when passed an invalid template.
+ * @type {String}
+ */
+const INVALID_TEMPLATE_TYPE_ERROR = 'Templates must be functions, instead got:';
+
+/**
+ * The error message logged when passed an unknown template name.
+ * @type {String}
+ */
+const UNKNOWN_TEMPLATE_NAME_ERROR = 'Templates must be configured, but the following template name is not:';
+
+/**
+ * The default transformer function that will be used to transform the tokens
+ * array into a string.
+ * @type {Function}
+ */
+const DEFAULT_TEMPLATES_TRANSFORMER = (tokens) => tokens.join('');
+
+/**
+ * Finds all template patterns and wraps them with the template (component)
+ * that matches its index, while cleaning up the templates' declaration
+ * symbols.
+ * @param {Object} options
+ * @param {String} options.originTranslation The translation into which the
+ *     templates will be injected.
+ * @param {Record.<String, Function>} options.templates The templates that will
+ *     be injected.
+ * @param {Function} options.templatesTransformer The templates transformer
+ *     function
+ * @return {String}
+ */
+const injectTemplates = ({ originTranslation, templates = {}, templatesTransformer = DEFAULT_TEMPLATES_TRANSFORMER }) => {
```

Comment:
It's not clear to me why do we need the `templatesTransformer`

Reply:
Responded in [the PR](https://github.com/fiverr/i18n-react_package/commit/2e3b53493c9bff6e76c3c7b79dc38372d1551f18#r124758552) for `i18n-react`